internal/datasource/os/files.go

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package os
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"path"
+
+	"github.com/nginx/agent/v3/api/grpc/instances"
+)
+
+func WriteFile(fileContent []byte, filePath string) error {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		slog.Debug("File does not exist, creating new file", "file", filePath)
+		err = os.MkdirAll(path.Dir(filePath), 0o750)
+		if err != nil {
+			return fmt.Errorf("error creating directory, directory: %v, error: %v", path.Dir(filePath), err)
+		}
+	}
+
+	err := os.WriteFile(filePath, fileContent, 0o644)
+	if err != nil {
+		return fmt.Errorf("error writing to file, filePath: %v, error:%v", filePath, err)
+	}
+	slog.Debug("Content written to file", "filePath", filePath)
+
+	return nil
+}
+
+func ReadInstanceCache(cachePath string) (map[string]*instances.File, error) {
```

Comment:
What's the reason why we are using a `map[string]*instances.File` instead of introducing a `struct` with `path` and `file` fields/attributes? As a caller of this code, I won't know that the key to this map represents a file path without looking at the unit tests for this file.

Reply:
Discussed this with Olli and Donal and decided to use an Alias for `map[string]*instances.File ` so its clearer what the map is